testing-common/src/main/java/org/hypertrace/agent/testing/AbstractHttpClientTest.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright The Hypertrace Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hypertrace.agent.testing;
+
+import io.opentelemetry.javaagent.instrumentation.api.Pair;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import org.hypertrace.agent.core.instrumentation.HypertraceSemanticAttributes;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public abstract class AbstractHttpClientTest extends AbstractInstrumenterTest {
+
+  protected static final TestHttpServer testHttpServer = new TestHttpServer();
+
+  private final boolean hasResponseBodySpan;
```

Comment:
think about the order of attributes: 
1. primitive types constants `static final`
2. other constants e.g. test server
3. class attributes
4. constructor